core/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -467,9 +468,14 @@ protected void doXContentAnalyzers(XContentBuilder builder, boolean includeDefau
             if (includeDefaults) {
                 builder.field("analyzer", "default");
             }
-        } else if (includeDefaults || fieldType().indexAnalyzer().name().startsWith("_") == false && fieldType().indexAnalyzer().name().equals("default") == false) {
-            builder.field("analyzer", fieldType().indexAnalyzer().name());
-            if (fieldType().searchAnalyzer().name().equals(fieldType().indexAnalyzer().name()) == false) {
+        } else {
+            if (includeDefaults || fieldType().indexAnalyzer().name().startsWith("_") == false
+                    && fieldType().indexAnalyzer().name().equals("default") == false) {
+                builder.field("analyzer", fieldType().indexAnalyzer().name());
+            }
+            if (includeDefaults || fieldType().searchAnalyzer().name().startsWith("_") == false
+                    && fieldType().searchAnalyzer().name().equals("default") == false
```

Comment:
But in that case we would not write search_analyzer in the case where the index_analyzer is `default` which would mean that the fix would not work (note the checks on this second if statement are checking the searchAnalyzer not the indexAnalyzer)